aten/src/ATen/cpu/vec/vec_half.h

```diff
@@ -0,0 +1,40 @@
+#include <ATen/cpu/vec/intrinsics.h>
+namespace at {
+namespace vec {
+// See Note [CPU_CAPABILITY namespace]
+inline namespace CPU_CAPABILITY {
+
+#if defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)
+inline uint16_t float2half_scalar(float val) {
```

Comment:
Why introduce these new functions that are only defined under both of these options?

It looks like they're only used once.

Reply:
There is no direct data type conversion instruction for single Half value on CPU, so I add scalar conversion with AVX instructions for Half to speed up when AVX2 or AVX512 is supported on the platform.